internal/gatewayapi/resource/load.go

```diff
@@ -315,6 +315,9 @@ func loadKubernetesYAMLToResources(input []byte, addMissingResources bool) (*Res
 		case KindSecret:
 			typedData := data.Interface()
 			typedStringData := stringData.Interface()
+			secretType := kobjVal.FieldByName("Type")
+			typedSecretType := secretType.Interface()
```

Comment:
I think we need to check if value [`IsValid()`](https://pkg.go.dev/reflect#Value.IsValid) before converting to an interface to avoid panics. Good to have a test with empty type as well.
```
typedSecretType := "Opaque" // default secret type
secretType := kobjVal.FieldByName("Type")
if secretType.IsValid() {
	typedSecretType := secretType.Interface()
}
```

Reply:
Thanks for the feedback!

`Good to have a test with empty type as well.`
Well, every already existing test case  has no secret type and the tests run without issue.

`I think we need to check if value IsValid() before converting to an interface to avoid panics`
I have changed the code according your advise... still, the test cases with no secret type do not show up the "Opaque" default type. For this reason I come to the conclusion that the valid check is not needed: as being ensured that the kind is "KindSecret" it is always given that "type" is an existing field that can be accessed. Otherwise, accessing other fields (like data or stringData) wouldn't be safe either.

Reply:
Ok I see the default might be an empty string, which is still a valid value. In that case can we could just do:
```suggestion
			secretType := kobjVal.FieldByName("Type").Interface()
```
But good to have the validation IMO. Either works for me.

Reply:
Variable name fixed, see ec05f63d008dc9980634b34f3c27090d11f61ec0. Thanks!